src/components/icon.tsx

```diff
@@ -0,0 +1,24 @@
+import type { ComponentType } from 'react';
+import type { StyleProp, ViewStyle } from 'react-native';
+import type { SvgProps } from 'react-native-svg';
+
+type Props = {
+  svg: ComponentType<SvgProps>;
+  size?: number;
+  color: string;
+  styleProps?: StyleProp<ViewStyle>;
+};
+
+const Icon = ({ svg, color, size = 30, styleProps = {} }: Props) => {
```

Comment:
Let's consider `24` as default `size` value
```suggestion
const Icon = ({ svg, color, size = 24, styleProps = {} }: Props) => {
```